### ISSUE TITLE
GCP Cloud Functions -> Cloud Build concurrency lost after v3.112.0

### ISSUE THREAD

**Issue by stakauskas** [pulumi/pulumi#15993]
### What happened?

Somewhere after versions of: `v3.112.0` Pulumi lost concurrency when deploying GCP Cloud Functions.

### Example

With version of: `pulumi-version: v3.112.0` if Cloud Functions deployed to GCP it would kick all of them at once and multiple build jobs would appear in Cloud Build.

With version of: >= `pulumi-version: v3.113.0` it would do them sequentially. Would start one build job, wait until finishes, then another job.

### Output of `pulumi about`

pulumi-version: v3.112.0 - works fine, builds in parallel
>= pulumi-version: v3.113.0 - builds spawned to Cloud Build are sequential.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by tgummerer** [comment]
This looks like a duplicate of https://github.com/pulumi/pulumi/issues/15991.  I'll close this out, please follow that issue.
